### ISSUE TITLE
Not working with mongoid embedded relationships

### ISSUE THREAD

**Issue by Bilal-Abbas** [DavyJonesLocker/client_side_validations#773]
I have embedded relation of address as "to_address" and "from_address" with shipment model
In shipment model as
`embeds_one  :to_address, class_name: "Address", as: :addressable`
`embeds_one  :from_address, class_name: "Address", as: :addressable`
and in address model as
`embedded_in :addressable, polymorphic: true`

but it is not working with" f.fields_for :to_address" and  "f.fields_for :from_address" as your documentation says...!

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the CSV Community.

> but it is not working with" f.fields_for :to_address" and "f.fields_for :from_address" as your documentation says...!

The documentation here is about activerecord, sorry if it is misleading

We have an [extension for mongoid](https://github.com/DavyJonesLocker/client_side_validations-mongoid), but it is only for presence and uniqueness validator. I'm using CSV, embedded documents, and mongoid in https://github.com/diowa/icare, but at the moment I do not recall if embedded validations were working or if I'm using those. You can take a look at the code

I'm maintaining CSV, CSV Simple Form, and CSV Mongoid in my free time, I will take a look in the weekend

Meanwhile, if you want to help me in speeding up things, you can setup and share on GitHub a minimal application with a reproducible test case

**Comment by tagliala** [comment]
Hi again,

thanks for your patience.

As I mentioned before, debugging and helping with CSV is a time consuming issue.

Please add `client-side-validations-mongoid` to your gems, I've just released a new version which supports CSV 16.0. It is only needed to provide support for the presence validation string

Going back to the issue you have reported, I can confirm that the nested objects validations work both in Active Record and in Mongoid, **as far as the associated object exists**

Example

```rb
# app/models/person.rb
class Person
  include Mongoid::Document

  embeds_one :address
  accepts_nested_attributes_for :address

  field :name, type: String

  validates :name, presence: true
end

# app/models/address.rb
class Address
  include Mongoid::Document

  embedded_in :person

  field :city, type: String

  validates :city, presence: true
end

# app/controllers/people_controller.rb (relevant code)

  # ...

  # GET /people/new
  def new
    @person = Person.new
    @person.build_address unless @person.address?
  end

    # ...

    def person_params
      params.require(:person).permit(:name, address_attributes: [:city])
    end

    # ...
```


```slim
# app/views/people/_form.html
= simple_form_for @person, validate: true do |f|
  = f.input :name

  = f.fields_for :address do |address_form|
    = address_form.input :city

  .actions = f.submit 'Submit', class: 'btn btn-primary'
```

Please note `@person.build_address unless @person.address?` in the controller: this ensures that the object exists.

I'm using:

- Ruby 2.6.5
- Rails 6
- Mongoid 7.0.5
- Simple Form 5.0.1
- CSV 16.0.3
- CSV Simple Form 9.1.0
- CSV Mongoid 9.1.0

![image](https://user-images.githubusercontent.com/556268/66705519-21059700-ed28-11e9-8c2e-ec661fb684ef.png)

Hope it helps

Closing here
